### ISSUE TITLE
Support for Apple Testflight

### ISSUE THREAD

**Issue by renep** [openbakery/gradle-xcodePlugin#98]
Add support that build for Apple TestFlight are automatically uploaded.

With iTMSTransporter the App can be uploaded to Apple: https://devforums.apple.com/message/1049285

**Comment by icecrystal23** [comment]
Is it still possible to upload to "old testflight"?

**Comment by renep** [comment]
In my current development branch I have removed this code.
I have read that rejects app now apps that includes the TestFlight SDK.

**Comment by icecrystal23** [comment]
I see... we don't use the SDK, though, so I guess I'll just restore testflight support in my fork until Apple kills the service altogether

**Comment by renep** [comment]
I have restored the 'old' testflight tasks for now

**Comment by saravnandm** [comment]
Hi renep,
Can you please share the example to upload to **apple test flight**? In the readme, I found it's supported. But nowhere its mentioned in the open bakery doc.

**Comment by renep** [comment]
There is an `appstoreUpload` and an `appstoreValidate` target, that takes the `ipa` that was created by a previous executed `package` task.
The appstore tasks needs only the appstoreconnect username and password so you have to set the `appstore.username` and `appstore.password` properties in the build.gradle file:

```
xcodebuild {
  ...
}

appstore {
  username = 'me@example.com' // your apple ID
  password = 'my apple id password'
}
```

If want to get asked for the password you can add this:
```
xcodebuild {
  ...
}
appstore.username = 'me@example.com' // your apple ID


gradle.taskGraph.whenReady { taskGraph ->
   if (taskGraph.hasTask(appstoreUpload) || taskGraph.hasTask(appstoreValidate)) {
     if (appstore.password == null) {
       def password = System.console().readPassword("\nPlease enter the appstore password for <" + appstore.username + ">: ")
       appstore.password = new String(password)
     }
   }
}
```
You get not ask if you specify the password with `-Pappstore.password=` from the command line when you use this. 

Note: You have to run gradle no as daemon, otherwise the shell input does not work e.g.:
` ./gradlew appstoreValidate --no-daemon`

**Comment by saravnandm** [comment]
Getting the below error for this **./gradlew appstoreValidate --no-daemon**

*** Error: Unable to validate your application. Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com
15:02:09.414 [QUIET] [org.openbakery.appstore.AbstractAppstoreTask] 2019-04-10 15:02:09.379 altool[17221:379829] *** Error: Unable to validate your application. Sign in with the app-specific password you generated. If you forgot the app-specific password or need to create a new one, go to appleid.apple.com
15:02:03.618 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
15:02:03.618 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :appstoreValidate FAILED

ALSO, CAN YOU CONFIRM THIS 'Appstore' will upload it to test flight only? if so what is the procedure to upload it to Prod app store?

**Comment by renep** [comment]
Looks like the appstore username and password is no accepted by the appstoreconnect portal. The error message that you see comes from the altool that is used to upload/validate the ipa. the altool is part of xcode.

There no difference between the upload for the TestFlight test and for upload for the Appstore production release. If you release your app in the App-Store you must first manually assign the (previously uploaded build) in the appstore connect portal and then submit it for review.

**Comment by saravnandm** [comment]
Any idea why it is not accepting? the account has second factor enabled. is it because of that?

**Comment by saravnandm** [comment]
Cool. The second factor caused that error. I have tried with different admin account which doesn't have the second factor enabled. **It's working now.**

Now I have a different problem,
1. After uploaded its was showing status in **processing**
2. After a few minutes that build got removed automatically from the app store connect. Why?

**One more question related to app size.**
How to set the behavior of the archiving/packaging, specifically wrt app thinning, stripping symbols, building symbol maps, etc.?

**Comment by renep** [comment]
Invalid Swift Support should be fixed with 0.15.4
(I have not uploaded a build yet with Xcode 10.2, because in progress of migrating to Xcode 10.2)

The gradle build includes the symbol maps in the ipa.
You can specify a `debug` or `release` build, and in the `release` build there should be no debug symbols.
For app thinging there is a `xcodebuild.bitcode` boolean option.

**Comment by saravnandm** [comment]
Yep your right it works with 0.15.4. Thanks renep :)
